File: iosApp/iosApp/Game/Game/GameSettings/GameSettingsScreen.swift
import SwiftUI
import shared

struct GameSettingsScreen: View {
	let gameSettings: GameSettings
	let onClose: () -> Void
	
	private let yes = "✅"
	private let no = "❌"
	
	var body: some View {
		VStack(alignment: .center, spacing: 16) {
			title
			settings
			Spacer()
			closeButton
		}
		.padding(.horizontal, 16)
	}
	
	@ViewBuilder
	private var title: some View {
		Text("game_settings")
			.font(.headline)
			.padding(16)
	}
	
	@ViewBuilder
	private var settings: some View {
		VStack(spacing: 16) {
			LabeledContent("finish_with_doubles", value: gameSettings.isFinishWithDoublesChecked ? yes : no)
			LabeledContent("turn_limit", value: gameSettings.isTurnLimitEnabled ? yes : no)
			LabeledContent("randomize_player_order", value: gameSettings.isRandomPlayersOrderChecked ? yes : no)
			LabeledContent("enable_statistics", value: gameSettings.isStatisticsEnabled ? yes : no)
		}
	}
	
	@ViewBuilder
	private var closeButton: some View {
		Button("return_to_game", action: onClose)
			.padding(16)
	}
}

File: androidApp/src/main/java/com/alextos/darts/android/game/game/game_settings/GameSettingsScreen.kt
package com.alextos.darts.android.game.game.game_settings

import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.padding
import androidx.compose.material.Button
import androidx.compose.material.MaterialTheme
import androidx.compose.material.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.res.stringResource
import androidx.compose.ui.unit.dp
import com.alextos.darts.android.R
import com.alextos.darts.game.domain.models.GameSettings

@Composable
fun GameSettingsScreen(
    gameSettings: GameSettings,
    onClose: () -> Unit
) {

    Column(horizontalAlignment = Alignment.CenterHorizontally) {
        Title()
        Settings(gameSettings = gameSettings)
        CloseButton(onClose = onClose)
    }
}

@Composable
private fun Title() {
    Text(
        modifier = Modifier.padding(16.dp),
        text = stringResource(id = R.string.game_settings),
        style = MaterialTheme.typography.h3
    )
}

@Composable
private fun Settings(gameSettings: GameSettings) {
    val yes = "✅"
    val no = "❌"
    LabeledText(
        label = stringResource(id = R.string.finish_with_doubles),
        text = if (gameSettings.isFinishWithDoublesChecked) yes else no
    )
    LabeledText(
        label = stringResource(id = R.string.turn_limit),
        text = if (gameSettings.isTurnLimitEnabled) yes else no
    )
    LabeledText(
        label = stringResource(id = R.string.randomize_player_order),
        text = if (gameSettings.isRandomPlayersOrderChecked) yes else no
    )
    LabeledText(
        label = stringResource(id = R.string.enable_statistics),
        text = if (gameSettings.isStatisticsEnabled) yes else no
    )
}

@Composable
private fun LabeledText(label: String, text: String) {
    Row(
        modifier = Modifier
            .fillMaxWidth()
            .padding(horizontal = 16.dp, vertical = 8.dp),
        verticalAlignment = Alignment.CenterVertically,
        horizontalArrangement = Arrangement.SpaceBetween
    ) {
        Text(text = label)
        Text(text = text)
    }
}

@Composable
private fun CloseButton(onClose: () -> Unit) {
    Button(
        modifier = Modifier.padding(16.dp),
        onClick = onClose
    ) {
        Text(text = stringResource(id = R.string.return_to_game))
    }
}